### PR TITLE
reworked examples

### LINKED ISSUE #161: example notebooks
I have gone through the examples and made a few changes, which I would like to submit a pull request for:
- in one of the datasets, the first row was accidentally used as a header
- in some cases, the label encoding was overwriting continuous variables with few unique values
- I have added a few helper functions for detecting categorical variables, getting their column index and getting embedding dimensions
- short benchmark on different embedding sizes for Adult Census Income dataset
- added comparison for LightGBM next to XGBoost
- added parts to compare feature importance from TabNet, XGBoost and LightGBM
- added links to open notebooks in Google Colab in the readme

### PR DESCRIPTION
**Pull request by frankherfert**
Improve the example notebooks.

**What kind of change does this PR introduce?**
- added new dataset for example notebooks
- moved examples into their own folder
- in one of the datasets, the first row was accidentally used as a header
- in some cases, the label encoding was overwriting continuous variables with few unique values
- I have added a few helper functions for detecting categorical variables, getting their column index and getting embedding dimensions
- short benchmark on different embedding sizes for Adult Census Income dataset
- added comparison for LightGBM next to XGBoost
- added parts to compare feature importance from TabNet, XGBoost and LightGBM
- added links to open notebooks in Google Colab in the readme

**Does this PR introduce a breaking change?**
- no

**What needs to be documented once your changes are merged?**
- readme file now contains links to open notebooks in Google colab

**Closing issues**
- closes #161
- has some overlap to #158

### CODE REVIEW

**Comment by frankherfert** [comment]
The links would work after the PR is merged. Google Colab supports loading notebooks from github directly.

I didn't know about the CI part.
Can we just create a new /examples folder and keep the original notebooks intact?
If you don't want the new notebooks in the repository, I can just keep them on my own profile and create a public link, but having them in the repository would be best for new users and future PRs in my opinion.

**Comment by kmedved** [comment]
I would find additional examples extremely helpful, so to the extent they can be merged, I think this would be a great help in getting people up to speed on a new tabular data approach.

**Comment by frankherfert** [comment]
I think it's a good idea if the other example pull request is merged first, then I will create a new one with the new notebooks for the new datasets and make sure they don't affect the CI tests.
Can you let me know if this works for you or if you would prefer a different approach?

**Comment by Optimox** [comment]
@frankherfert Let's merge first the other PR and then I think we can create an example folder to add your contributions.

**Comment by Optimox** [comment]
@frankherfert I'm closing this as there has been some important changes in the way TabNet works so this is not up to date.
